flash/vision/classification/data.py

```diff
@@ -211,7 +211,7 @@ def __init__(
     ) -> 'ImageClassificationData':
         """Creates a ImageClassificationData object from lists of image filepaths and labels"""
 
-        if train_dataset is not None and train_split is not None or val_split is not None or test_split is not None:
+        if train_dataset is not None and train_split is not None or val_split is not None and test_split is not None:
```

Comment:
@tchaton I'm just noticing that this fix doesn't solve at al the issues. This prevents of not raising an error of `len(dataset)==0` but later the validation stage it's triggered.